pvlib/financial.py

```diff
@@ -0,0 +1,157 @@
+import numpy as np
+
+
+def lcoe(production=None, cap_cost=None, fixed_om=None):
+    """
+    Real levelized cost of electricity (LCOE).
+
+    Includes cost of capital and fixed operations and maintenance (O&M), but
+    not variable O&M. Described in [1]_, pp. 43 and 47-48, and defined here as
```

Comment:
something I learned recently is that the variable o&m cost is zero for solar and typ. refers to the cost of fuel for fossil fuel power

Reply:
also, if I understand correctly both the simple and fixed charge rate LCOE models assume that the degradation rate is the same as the discount rate or the cost of capital, which is probably conservative. Deg rates are typically less than 1% while the suggested discount rate is 3%. The FCR simplification is:
```
LCOE = (FCR*CAPEX + OPEX_FIX) / YIELD + OPEX_VAR  # OPEX_VAR = zero for solar!
```
which is equivalent to: 
```
LCOE = (CAPEX + OPEX/FCR) / (YIELD/FCR)  # just use OPEX for fixed O&M
```
I believe we can do better than this by just using either the lifetime production in the denominator or using a different FCR for production:
```
LCOE = (CAPEX + OPEX/FCR) / (YIELD/FIXDEGRATE)
# where FIXDEGRATE = [1 / SUM(1 / (1+RD)**n) for n in range(N)]
# N is lifetime in years, same as used for fixed charge rate
# and RD is the degradation rate, usually 0.75%/year
```
I think this might return an LCOE with more realistic accounting for degradation?

Reply:
The LCOE models apply the financial discount rate to the future energy, see equation 4-6 of https://www.nrel.gov/docs/legosti/old/5173.pdf

PV degradation needs to be included in the yield number. For a discounted cash flow we use equation 4-6 (the sum with specific values for each year). For the simple LCOE formula the ATB uses an average capacity factor over the life of the system.

Reply:
I didn’t realize the FCR applies to the energy as well, I guess it makes sense that it’s value changes over time because of inflation.

okay so then the average yield should be the total production including degradation divided by the lifetime of the system? Thanks!

Reply:
> okay so then the average yield should be the total production including degradation divided by the lifetime of the system? Thanks!

That's correct!

Reply:
Not to dampen enthusiasm, but does the reference include degradation? I know that including it may give a more meaningful LCOE, but here, we're implementing what's been published. The function can be extended or modified to include degradation but I'd like to see a second reference describing how that's done.

Reply:
@eccoope I think there should be something in the docs or example that shows that the fuel cost is zero for solar. Perhaps other newbies to financing like me will also be confused

Reply:
> Not to dampen enthusiasm, but does the Short et al. reference include degradation? I know that including degradation may give a more meaningful LCOE, but here, we're implementing what's been published. The function can be extended or modified to include degradation but I'd like to see a second reference describing how that's done.

Looking at the code again, given that production is an array, it might be better to have users include degradation in that array. In that case, equation 4-6 for Short et al. still works. If you prefer a year 1 energy yield and a degradation factor, equation 2 of https://pubs.rsc.org/en/content/articlehtml/2011/ee/c0ee00698j provides a really good example.
 

If there is demand for the average yield approach I can ping the PV team for the ATB.

Reply:
Hi @mikofski ! Degradation rates are implemented as an average capacity factor from the ATB as mentioned by @brtietz in [simple_lcoe_calculator.py](https://github.com/pvlib/pvlib-python/pull/1687/files#diff-fbb6aa03b751a221735e8c7a78a86a5d68657ff3b525f6f40f02ab37be8951f4) (but I could probably be more explicit about this in a comment), but there's also a discount rate of 3% in this example which is separate from degradation.

Degradation is not accounted for in [docs/examples/financial/lcoe_sam_validation.py](https://github.com/pvlib/pvlib-python/pull/1687/files#diff-c033013cbda2f090525a4ffc4ee64f3249311d96bb4fcbb22450a564033ff80d) - constant AC power is explicitly assumed (see line 126) - but we could easily incorporate by multiplying the production array element-wise with a degradation array like the one you defined as `FIXEDEGRATE`.

@mikofski and @brtietz - if either of you would be interested in meeting with me and possibly @cwhanse for 30 minutes to clarify any of these details, let me know

Reply:
@eccoope Sure - please email me at brian.mirletz (at) nrel (dot) gov to set up a time.